### PR TITLE
feat: ForIn instance for NameMap and PersistentHashMap

### PR DESCRIPTION
**Pull request by digama0**
Implements some missing `ForIn` instances. For `PersistentHashMap` I started to do a manual implementation but it turns out the auxiliary function needs to bubble up the `done` error so it's the same as just using `foldlM` directly on the `ExceptT σ m` monad. So this implementation just delegates to that.

### CODE REVIEW

File: src/Lean/Parser/Basic.lean [hunk 1]
```diff
@@ -1610,7 +1612,8 @@ inductive LeadingIdentBehavior where
   The method `termParser prec` is equivalent to the method above.
 -/
 structure ParserCategory where
-  tables   : PrattParsingTables
+  kinds    : SyntaxNodeKindSet := {}
```
Comment (leodemoura) [inline src/Lean/Parser/Basic.lean:1615]: This is change is not related to the PR.

Comment (digama0) [reply src/Lean/Parser/Basic.lean:1615]: oops
